### DISCUSSION TITLE
wallet_addEthereumChain/wallet_switchEthereumChain does not work properly in metamask mobile

### DISCUSSION THREAD

**Discussion by ghost** [WalletConnect/walletconnect-monorepo#2901] [Developer Support]
Currently I'm using WalletConnect with Wagmi to interact with Metamask mobile, I would like to implement RPC request to programatically add/switch network on Metamask mobile (wallet_addEthereumChain/wallet_switchEthereumChain) but it doesn't work properly.
Firstly, the document tell me to setup like this:
```
const chains = <MY-CUSTOM-NETWORK> as Chain[];
const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
export const wagmiClient = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({ chains, projectId }),
  publicClient,
});
const ethereumClient = new EthereumClient(wagmiClient, chains);
export const web3modal = new Web3Modal(
  {
    projectId,
    explorerExcludedWalletIds: 'ALL',
  },
  ethereumClient
);
```
Then I'm using `const ethereum = await wagmiClient.connector?.getProvider();` to interact with rpc request like this:
```
const network = <MY-CUSTOM-NETWORK>;
console.log('before');
await ethereum.request({
   method: 'wallet_addEthereumChain',
   params: [network],
});
console.log('after')
```

Is this a proper implementation? I've noticed that it worked normally on Chrome Metamask extension. On Metamask mobile, the request popup didn't show the first time, when I logged out, then logged in again, and sent a second request, the popup showed. Then on the website, the request was stuck after I choose "Switch Network" on my Metamask mobile and doesn't show 'after' on the console for both first and second requests.

I'm using packages:
```
"@wagmi/core": "^1.3.3",
"@web3modal/ethereum": "^2.6.2",
"@web3modal/html": "^2.6.2",
"viem":"1.2.6",
```

**Comment by crypblizz8** [comment]
Hey there, MM will be coming out with a new fix for this in their upcoming release. Suggest testing with another wallet for now. Do you get the same error?
